modules/queue/queue_bytefifo.go

```diff
@@ -126,21 +142,23 @@ func (q *ByteFIFOQueue) readToChan() {
 			if err != nil {
```

Comment:
Just change `default` to a timer.C, after the time is out, try to Pop . This will make the code simpler.

Reply:
There is no backoff when there is a successful pop